### DISCUSSION TITLE
Multiple Editors in the same View

### DISCUSSION THREAD

**Discussion by nadar** [ueberdosis/tiptap#3088] [General Questions] [answered]
Hi TipTap Community

I wan't to ask if someone else had a problem (or successfully) setting up multiple editors on the same view. We have a huge problem when iterating editors and remove one of them. It will always remove the last editor in the iteration, even when the element in the middle should be removed.

We have made an issue and also codesandbox where its very easy to reproduce the problem. 

+ [Codesandbox, add 3 or more items and remove the one in the middle and i will always remove the latest tiptap editor instance](https://codesandbox.io/s/tiptap-vue-multiple-editor-issue-oxr1lo?file=/src/App.vue)
+ [Issue which further describes the Bug](https://github.com/ueberdosis/tiptap/issues/3067)

Does anyone has an idea of what we could try to fix this problem?

**Comment by calebfrieze** [comment]
I am also having the same issue

**Comment by justindantzer** [comment, marked as answer]
I believe the issue experienced was due to using the array index as the `key` for the items loop. When using ordered numerical indexes, when removing an item, the following item assumes the same index as the item removed. Vue will not update a nested component in this case.

If a more unique or random key is used, everything should work as expected.

[Example forked from the original sandbox](https://codesandbox.io/p/sandbox/tiptap-vue-multiple-editor-issue-forked-znxcph)

Relevant Vue docs: [Maintaining State with key](https://vuejs.org/guide/essentials/list#maintaining-state-with-key), [Key API](https://vuejs.org/api/built-in-special-attributes.html#key)
